### PR TITLE
feat : Setup empty modules and workflow

### PR DESCRIPTION
**Pull request by anarahari**
Empty modules have been setup 
data
domain
presentation
common

### CODE REVIEW

File: .idea/gradle.xml
Comment (shreyansh-mm) [inline .idea/gradle.xml:1]: This file should be in `.gitignore`, it should not be in VCS

File: app/build.gradle.kts [hunk 1]
```diff
@@ -4,7 +4,7 @@ plugins {
     alias(libs.plugins.kotlin.compose)
     alias(libs.plugins.apollo)
     alias(libs.plugins.kotlinter)
-    alias(libs.plugins.ksp)
+    //alias(libs.plugins.ksp)
```
Comment (shreyansh-mm) [inline app/build.gradle.kts]: Let's uncomment it or remove it.

**Review by shreyansh-mm — CHANGES_REQUESTED** [review]
Let's add a workflow which will do lint checks and run all the unit tests for you on every PR.

File: common/build.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+}
+
+dependencies {
+    implementation(libs.androidx.core.ktx)
+    implementation(libs.androidx.appcompat)
+    implementation(libs.material)
+    androidTestImplementation(libs.androidx.espresso.core)
+    implementation(libs.apollo.runtime)
+    implementation(libs.dagger)
+    //ksp(libs.dagger.compiler)
```
Comment (shubhamsinghmutualmobile) [inline common/build.gradle.kts]: Same here (let's remove all unused code)

File: data/build.gradle.kts [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+dependencies {
+    implementation(project(":domain"))
+    implementation(project(":common"))
+    implementation(libs.androidx.core.ktx)
+    implementation(libs.androidx.appcompat)
+    implementation(libs.material)
+    androidTestImplementation(libs.androidx.espresso.core)
+    implementation(libs.apollo.runtime)
+    implementation(libs.dagger)
+    //ksp(libs.dagger.compiler)
```
Comment (shubhamsinghmutualmobile) [inline data/build.gradle.kts]: Same

File: domain/build.gradle.kts [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    implementation(project(":common"))
+    implementation(libs.androidx.core.ktx)
+    implementation(libs.androidx.appcompat)
+    implementation(libs.material)
+    androidTestImplementation(libs.androidx.espresso.core)
+    implementation(libs.apollo.runtime)
+    implementation(libs.androidx.paging.compose)
+    implementation(libs.androidx.paging.runtime)
+    implementation(libs.dagger)
+    //ksp(libs.dagger.compiler)
```
Comment (shubhamsinghmutualmobile) [inline domain/build.gradle.kts]: Same

File: presentation/build.gradle.kts [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+    implementation(project(":domain"))
+    implementation(libs.androidx.core.ktx)
+    implementation(libs.androidx.appcompat)
+    implementation(libs.material)
+    implementation(libs.androidx.material3.android)
+    androidTestImplementation(libs.androidx.espresso.core)
+    implementation(platform(libs.androidx.compose.bom))
+    implementation(libs.androidx.navigation.compose)
+    implementation(libs.dagger)
+    //ksp(libs.dagger.compiler)
```
Comment (shubhamsinghmutualmobile) [inline presentation/build.gradle.kts]: Same

**Comment by anarahari** [comment]
> Let's add a workflow which will do lint checks and run all the unit tests for you on every PR.

added the same

**Review by shreyansh-mm — APPROVED** [review]

**Pull request merged**